source/dynamical_systems/src/DynamicalSystem.cpp

```diff
@@ -31,15 +32,30 @@ bool DynamicalSystem<state_representation::CartesianState>::is_compatible(const
       && state.get_reference_frame() != this->get_base_frame().get_reference_frame());
 }
 
+template<>
+bool DynamicalSystem<state_representation::JointState>::is_compatible(const state_representation::JointState& state) {
+  return (state.get_name() == this->get_base_frame().get_name()
```

Comment:
Does this makes sense for `JointState`? Because it means you need to have a base frame which I am not sure what would be the usage. I think it actually would make sense to check compatibility with the attractor, in that case putting the function `virtual` and `overriding` it in the base classes. You could also use directly the `is_compatible` function of `JointState` actually.

Reply:
I guess if your derived class constructors correctly initialize it then it should be okay. It would not exactly be a "base frame" in that case but more like an accepted state.

Reply:
Good point and one that was destined to come up. Indeed for the JointState it doesn't really make sense, I just though of it as a place holder for the name and joint_names because the base_frame_ is required to be set in the base class. The derived class constructors initialize them with the same name and joint_names as the attractor, but it could technically be changes by the user with `set_base_frame`. So I'm not sure what the best solution is here.

Reply:
I think it makes sense - while we don't have the same reference frame logic as in Cartesian space, we still expect the robot and joint names to be the same in order to be compatible. The base frame still has a meaning in this case, and it's up to the derived classes to set it according to the intended usage.